### PR TITLE
Feature/ support terra2 (WIP)

### PR DESCRIPTION
**Pull request by JSHan94**

### CODE REVIEW

File: terra_sdk/client/lcd/lcdclient.py [hunk 1]
```diff
@@ -37,12 +34,12 @@
 
 
 def get_default(chain_id: str) -> [Coins, Numeric]:
-    if chain_id == "columbus-5":
-        return [Coins.from_str("0.15uusd"), Numeric.parse(1.75)]
-    if chain_id == "bombay-12":
-        return [Coins.from_str("0.15uusd"), Numeric.parse(1.75)]
+    if chain_id == "pheonix-5":
```
Comment (Vritra4) [inline terra_sdk/client/lcd/lcdclient.py:37]: @JSHan94 fix this please :)

File: terra_sdk/core/compact_bit_array.py [hunk 2]
```diff
@@ -29,12 +29,6 @@ def to_data(self) -> dict:
             "elems": base64.b64encode(self.elems)
         }
 
-    def to_data(self) -> dict:
```
Comment (Vritra4) [inline terra_sdk/core/compact_bit_array.py:32]: @JSHan94 why is it removed?

File: terra_sdk/core/wasm/data.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    
+    def to_proto(self) -> AccessConfig_pb:
+        return AccessConfig_pb(
+            permission=self.permission,
+            address=self.address
+        )
+
+    @classmethod
+    def from_data(cls, data:dict) -> AccessConfig:
```
Comment (Vritra4) [inline terra_sdk/core/wasm/data.py:43]: @JSHan94 we need to convert from primitive types to AccessType and vice versa.

File: tests/client/lcd/api/feegrant_test.py [hunk 4]
```diff
@@ -14,11 +15,3 @@ def test_allowances():
     )
     assert result is not None
     assert len(result) == 0
-
-
-# def test_allowance():
```
Comment (Vritra4) [inline tests/client/lcd/api/feegrant_test.py:19]: @JSHan94 keep blocked and unblock after feegranting on chain.

File: terra_sdk/client/lcd/lcdclient.py (hunk 1 above)
Comment (JSHan94) [reply terra_sdk/client/lcd/lcdclient.py:37]: oh, got it. thanks :)

File: terra_sdk/core/compact_bit_array.py (hunk 2 above)
Comment (JSHan94) [reply terra_sdk/core/compact_bit_array.py:32]: same function declared twice. see 26-31 lines

File: terra_sdk/core/wasm/data.py (hunk 3 above)
Comment (JSHan94) [reply terra_sdk/core/wasm/data.py:43]: yes. actually, core/wasm is mostly WIP

**Comment by JSHan94** [comment]
Done :
- pass all testcases in /tests folder

Todo :
- wasmAPI fix (history, absolutetx)
- createTx error (currently, only work in MsgSend)

**Comment by JSHan94** [comment]
- Support only Terra 2.0 (no isClassic flag)
- Basic test cases (including integration-tests) works but not fully tested yet. Especially, ibc module should be tested
- Docs should be updated (WIP)

File: integration_tests/contract.py [hunk 5]
```diff
@@ -1,10 +1,13 @@
 from pathlib import Path
 
+from terra_proto.cosmwasm.wasm.v1 import AccessType
```
Comment (Vritra4) [inline integration_tests/contract.py:3]: we should re-export AccessType from terra.proto

**Review by hanjukim — APPROVED** [review]

File: terra_sdk/client/lcd/lcdclient.py [hunk 6]
```diff
@@ -37,12 +34,12 @@
 
 
 def get_default(chain_id: str) -> [Coins, Numeric]:
-    if chain_id == "columbus-5":
-        return [Coins.from_str("0.15uusd"), Numeric.parse(1.75)]
-    if chain_id == "bombay-12":
-        return [Coins.from_str("0.15uusd"), Numeric.parse(1.75)]
+    if chain_id == "pheonix-1":
```
Comment (Vritra4) [inline terra_sdk/client/lcd/lcdclient.py:37]: phoenix-1 is our mainnet. :)

**Pull request merged**